src/main/java/com/felixgrund/codeshovel/parser/impl/JavaParser.java

```diff
@@ -1 +1,111 @@

import java.util.ArrayList;
import java.util.List;

public class JavaParser extends AbstractParser implements Yparser {

	public static final String ACCEPTED_FILE_EXTENSION = ".java";
	private Logger log = LoggerFactory.getLogger(JavaParser.class);

	public JavaParser(StartEnvironment startEnv, String filePath, String fileContent, Commit commit) throws ParseException {
		super(startEnv, filePath, fileContent, commit);
	}

	@Override
	protected List<Yfunction> parseMethods() throws ParseException {
		CompilationUnit rootCompilationUnit = com.github.javaparser.JavaParser.parse(this.fileContent);
		if (rootCompilationUnit == null) {
			throw new ParseException("Could not parseMethods root compilation unit", this.filePath, this.fileContent);
		}
		JavaMethodVisitor visitor = new JavaMethodVisitor() {
			@Override
			public boolean methodMatches(Yfunction method) {
				return method.getBody() != null;
			}
		};
		rootCompilationUnit.accept(visitor, null);
		return visitor.getMatchedNodes();
	}

	@Override
	public double getScopeSimilarity(Yfunction function, Yfunction compareFunction) {
		return Utl.parentNamesMatch(function, compareFunction) ? 1.0 : 0.0;
	}

	@Override
	public String getAcceptedFileExtension() {
		return ACCEPTED_FILE_EXTENSION;
	}

	@Override
	public List<Ychange> getMinorChanges(Ycommit commit, Yfunction compareFunction) {
		List<Ychange> changes = new ArrayList<>();
		Yreturntypechange yreturntypechange = getReturnTypeChange(commit, compareFunction);
		Ymodifierchange ymodifierchange = getModifiersChange(commit, compareFunction);
		Yexceptionschange yexceptionschange = getExceptionsChange(commit, compareFunction);
		Ybodychange ybodychange = getBodyChange(commit, compareFunction);
		Yparametermetachange yparametermetachange = getParametersMetaChange(commit, compareFunction);
		if (yreturntypechange != null) {
			changes.add(yreturntypechange);
		}
		if (ymodifierchange != null) {
			changes.add(ymodifierchange);
		}
		if (yexceptionschange != null) {
			changes.add(yexceptionschange);
		}
		if (ybodychange != null) {
			changes.add(ybodychange);
		}
		if (yparametermetachange != null) {
			changes.add(yparametermetachange);
		}
		return changes;
	}



	private Yfunction transformMethod(MethodDeclaration method) {
		return new JavaFunction(method, this.commit, this.filePath, this.fileContent);
	}

	public abstract class JavaMethodVisitor extends VoidVisitorAdapter<Void> {

		private List<Yfunction> matchedNodes = new ArrayList<>();

		public abstract boolean methodMatches(Yfunction method);

		@Override
		public void visit(MethodDeclaration method, Void arg) {
			super.visit(method, arg);
			boolean hasBody = method.getBody().isPresent();
			Yfunction yfunction = transformMethod(method);
			if (hasBody && methodMatches(yfunction)) {
				matchedNodes.add(yfunction);
			}
		}

		public List<Yfunction> getMatchedNodes() {
			return matchedNodes;
		}
	}

}
\ No newline at end of file
+package com.felixgrund.codeshovel.parser.impl; 
```

Comment:
Why is the diff so broken here? Wait. Did you change from tabs to spaces or something? Can you check in your IDE what your indent settings are?

Reply:
haha Felix this was the change you asked me to do months ago! For some reason it had the `^M` character on master, so I fixed it but it never got merged.

Reply:
Oh ok. As long as it's better afterwards. Just make it tabs everywhere plz. I've been fiddling with spaces vs. tabs just too often.
Check this out:  https://www.youtube.com/watch?v=SsoOG6ZeyUI